Add LiteralValue wrapper rendered without templating (#35017)
mistercrunch/Airflow: docs/apache-airflow/core-concepts/operators.rst

Some operators will also consider strings ending in specific suffixes (defined in ``template_ext``) to be references to files when rendering fields. This can be useful for loading scripts or queries directly from files rather than including them into DAG code.

For example, consider a BashOperator which runs a multi-line bash script, this will load the file at ``script.sh`` and use its contents as the value for ``bash_command``:

.. code-block:: python

    run_script = BashOperator(
        task_id="run_script",
        bash_command="script.sh",
    )

By default, paths provided in this way should be provided relative to the DAG's folder (as this is the default Jinja template search path), but additional paths can be added by setting the ``template_searchpath`` arg on the DAG.

> .. versionadded:: 2.8
>     In some cases, you may want to exclude a string from templating and use it directly. Consider the following task:

Reviewer: ```suggestion
In some cases, you may want to exclude a string from templating and use it directly. Consider the following task:
```

.. code-block:: python

    print_script = BashOperator(
        task_id="print_script",
        bash_command="cat script.sh",
    )

This will fail with ``TemplateNotFound: cat script.sh``, but we can prevent airflow from treating this value as a reference to a file by wrapping it in :func:`~airflow.util.template.literal`.
This approach disables the rendering of both macros and files and can be applied to selected nested fields while retaining the default templating rules for the remainder of the content.

.. code-block:: python

    fixed_print_script = BashOperator(
        task_id="fixed_print_script",
        bash_command=literal("cat script.sh"),
    )

> 

Reviewer: ```suggestion

.. versionadded:: 2.8
    :func:`~airflow.util.template.literal` was added.

```

Some operators will also consider strings ending in specific suffixes (defined in ``template_ext``) to be references to files when rendering fields. This can be useful for loading scripts or queries directly from files rather than including them into DAG code.

For example, consider a BashOperator which runs a multi-line bash script, this will load the file at ``script.sh`` and use its contents as the value for ``bash_command``:

.. code-block:: python

    run_script = BashOperator(
        task_id="run_script",
        bash_command="script.sh",
    )

By default, paths provided in this way should be provided relative to the DAG's folder (as this is the default Jinja template search path), but additional paths can be added by setting the ``template_searchpath`` arg on the DAG.

In some cases, you may want to exclude a string from templating and use it directly. Consider the following task:

.. code-block:: python

    print_script = BashOperator(
        task_id="print_script",
        bash_command="cat script.sh",
    )

> This will fail with ``TemplateNotFound: cat script.sh``, but we can prevent airflow from treating this value as a reference to a file by wrapping it in :func:`~airflow.util.template.literal`.

Reviewer: ```suggestion
This will fail with ``TemplateNotFound: cat script.sh`` since Airflow would treat the string as a path to a file, not a command.
We can prevent airflow from treating this value as a reference to a file by wrapping it in :func:`~airflow.util.template.literal`.
```

This approach disables the rendering of both macros and files and can be applied to selected nested fields while retaining the default templating rules for the remainder of the content.

.. code-block:: python

> 

Reviewer: ```suggestion

    from airflow.utils.template import literal

```

    fixed_print_script = BashOperator(
        task_id="fixed_print_script",
        bash_command="cat script.sh",
    )
    fixed_print_script.template_ext = ()

It is also possible to exclude a string from templating and use it directly by wrapping it in
:class:`~airflow.template.templater.LiteralValue`. This approach disables the rendering of both macros and files and
can be applied to selected nested fields, while retaining the default templating rules for the remainder of the content.

.. code-block:: python

    fixed_print_script = BashOperator(
        task_id="fixed_print_script",
        bash_command=LiteralValue("cat script.sh"),
    )

> 

Reviewer: It’d be a good idea to add a `.. versionadded` marker somewhere around here.

Also, [`airflow.template.templater` is not a public module](https://airflow.apache.org/docs/apache-airflow/stable/public-airflow-interface.html), and I’m not sure if we want to open it up (all its other contents are private); we may want to move the class somewhere else.

Author: I see. In that case, `airflow.utils.template` might be a better location. The `airflow.utils` package contains other classes used to configure operators, such as `TriggerRule`.

I could define a function there, instead of directly exposing `LiteralValue`, which would make it easier to change the implementation in the future.

For example:
```
from airflow.template.templater import LiteralValue

def literal_value(value: Any) -> LiteralValue:
    return LiteralValue(value)
```

    fixed_print_script = BashOperator(
        task_id="fixed_print_script",
        bash_command="cat script.sh",
    )

>     fixed_print_script.template_ext = ()

Reviewer: Let’s move the `literal` solution before this. We want to recommend it over this somewhat hacky workaround.
